### PR TITLE
Tpacket v3 10

### PR DESCRIPTION
**Pull request by regit**
Rework of #2032.

Modifications:
* Merge commit on block-size.
* Add a patch to use time() instead of GetTime()
* Add a patch to print mmap error
* Address comments of #2032

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/167
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/163

### CODE REVIEW

**Comment by inliniac** [comment]
https://redmine.openinfosecfoundation.org/issues/681

File: src/source-af-packet.c [hunk 1]
```diff
@@ -1388,14 +1587,223 @@ frame size: TPACKET_ALIGN(snaplen + TPACKET_ALIGN(TPACKET_ALIGN(tp_hdrlen) + siz
...
+    ptv->req.tp_frame_size = TPACKET_ALIGN(snaplen +TPACKET_ALIGN(TPACKET_ALIGN(tp_hdrlen) + sizeof(struct sockaddr_ll) + ETH_HLEN) - ETH_HLEN);
+    frames_per_block = ptv->req3.tp_block_size / ptv->req3.tp_frame_size;
+
+    if (frames_per_block == 0) {
+        SCLogError(SC_ERR_INVALID_VALUE,
+                   "Block size is too small, it should be at least %d",
+                   ptv->req3.tp_frame_size);
+        return -1;
+    }
+    ptv->req3.tp_block_nr = ptv->ring_size / frames_per_block + 1;
```
Comment (inliniac) [inline src/source-af-packet.c]: why is block_nr (which I interpret as: the number of blocks of size 'ptv->block_size') not calculated as ring_size / block_size?

Comment (regit) [inline src/source-af-packet.c]: ring_size is in number of elements not in bytes. So we divide by number of frame per block to get number of elements. This is for sure making less sense in v3 than in v2.

**Comment by regit** [comment]
Closing and resubmitting with rebase and a little fix.

**Pull request closed without merging**